src/python/pants/engine/subsystem/native.py

```diff
@@ -550,57 +599,81 @@ def visualize_to_dir(self):
     return self._visualize_to_dir
 
   @memoized_property
+  def binary(self):
+    """Load and return the path to the native engine binary."""
+    return self._binary_util.select_binary(self._supportdir,
+                                           self._version,
+                                           '{}.so'.format(NATIVE_ENGINE_MODULE))
+
+  @memoized_property
   def lib(self):
-    """Load and return the `libgraph` module."""
-    binary = self._binary_util.select_binary(self._supportdir,
-                                              self._version,
-                                              'native-engine')
-    return _FFI.dlopen(binary)
+    """Load and return the native engine module."""
+    return self.ffi.dlopen(self.binary)
+
+  @memoized_property
+  def ffi(self):
```

Comment:
the diff mightve cut off the class context, but there are two different classes here that are both holding a reference to the same `CompiledCFFI` object (the equivalent of `from _native_engine import ffi`):

- `Native` (the subsystem), which exposes the `self.ffi` property
- `ExternContext`, which gets a `self._ffi` attribute at creation time (which is just a reference to `Native.ffi`)

the loaded externs are in the `ffi_lib` property, which is the equivalent of `from _native_engine import lib` and is where the loaded externs hang off of (e.g. `lib.extern_invoke_runnable` would be the callable).